### PR TITLE
Governance migrate to subgraph and MVP

### PR DESCRIPTION
**Pull request by bearpong**
CHANGELOG:
- migrates to our subgraph, remoing tally
- adds useBlocktoTimestamp
- moves BeraJS Provider out of berawagmi, isolating the library in order to use it inside @bera/berajs
- filters proposal by topic in each page
- uses subgraph timelockid
- computes proposal status dynamically based on timestamps/blocks
- adds proposal filtering
- adds proposal votes filtering

### CODE REVIEW

File: packages/berajs/src/config/defaultBeraConfig.ts [hunk 1]
```diff
@@ -59,26 +59,8 @@ export const defaultBeraConfig: BeraConfig = {
...
-      berps: {
-        governor: governorAddress,
-        timelock: governanceTimelockAddress,
-      },
-      general: {
-        governor: governorAddress,
-        timelock: governanceTimelockAddress,
-      },
+      governor: governorAddress,
+      timelock: governanceTimelockAddress,
```
Comment (BrownBrownBear) [inline packages/berajs/src/config/defaultBeraConfig.ts:63]: Did homme approve that this is going to be from the config or was it from an external source like something we set for people. can people manipulate this time?

File: packages/graphql/src/modules/index.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 export * from "./types";
+
 export * from "./chain";
 export * from "./dex";
 export * from "./honey";
 export * from "./lend";
-export * from "./governance";
+// export * from "./governance";
```
Comment (BrownBrownBear) [inline packages/graphql/src/modules/index.ts:7]: was this because of the codegen and you don't need any types for these queries anymore?

File: packages/graphql/tsup.config.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
+import { readFile, writeFile } from "fs/promises";
+
+const entryPoints = [
+  "./src/modules/governance/index.ts",
+  "./src/modules/chain/index.ts",
+];
```
Comment (BrownBrownBear) [inline packages/graphql/tsup.config.ts:6]: I'll add other dapps and modules when we're confident with this codegen process in future PRs

File: packages/wagmi/package.json [hunk 4]
```diff
@@ -27,7 +27,6 @@
     "dev": "tsup src --watch"
   },
   "dependencies": {
-    "@bera/berajs": "workspace:*",
```
Comment (BrownBrownBear) [inline packages/wagmi/package.json:30]: why was this removed? just not needed i'm guessing and it's a circular dependency when it should be one direction?

**Review by BrownBrownBear — APPROVED** [review]
I've read through all your changes and for the most part everything seems good. Just reply to my comments and we can merge this PR. Let's not add anything more to this and then we can start breaking down smaller tasks and improvements based off this initial design. Good work on the migration, this was a lot of work that was needed. We can maybe merge this tomorrow.

File: packages/graphql/src/modules/index.ts (hunk 2 above)
Comment (bearpong) [reply packages/graphql/src/modules/index.ts:7]: @BrownBrownBear unfortunately codegen generates the same types for each schema, so that it would break the typescript build. There's probably some way to make it work properly but i opted for using the sub-package "@bera/graphql/governance" set in the package.json. In the future we can export every module individually.

File: packages/wagmi/package.json (hunk 4 above)
Comment (bearpong) [reply packages/wagmi/package.json:30]: yes, it'd cause a circular dependency. it was only used to include BeraJSProvider inside the WagmiProvider. I moved it out and added berajs provider to each dapp [in this commit](https://github.com/berachain/monobera/pull/117/commits/8a90c6308f966bd0133b36e93585772c9f33a7d4)

File: packages/berajs/src/config/defaultBeraConfig.ts (hunk 1 above)
Comment (bearpong) [reply packages/berajs/src/config/defaultBeraConfig.ts:63]: @BrownBrownBear what do you mean? There's one only governance contract for our dapps, why would one want to use another contract? Same for the timelock contract.
